### PR TITLE
fix: [CDS-44552]: Move YamlChangeSetPruneJob to Iterator Framework

### PR DESCRIPTION
**Pull request by abhinav-harness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/38738)

### CODE REVIEW

File: 360-cg-manager/src/main/java/software/wings/app/WingsApplication.java [hunk 1]
```diff
@@ -1485,7 +1484,7 @@ private void registerCronJobs(Injector injector) {
       // If we do not get the lock, that's not critical - that's most likely because other managers took it
       // and they will initialize the jobs.
       if (acquiredLock != null) {
-        YamlChangeSetPruneJob.add(jobScheduler);
+        //        YamlChangeSetPruneJob.add(jobScheduler);
```
Comment (meenakshi-harness) [inline 360-cg-manager/src/main/java/software/wings/app/WingsApplication.java:1487]: Remove

File: 400-rest/src/main/java/software/wings/yaml/gitSync/GitSyncEntitiesExpiryHandler.java [hunk 2]
```diff
@@ -55,16 +60,24 @@ public class GitSyncEntitiesExpiryHandler implements Handler<Account> {
...
 
   private static long ONE_MONTH_IN_MILLIS = 2592000000L;
   private static long TWELVE_MONTH_IN_MILLIS = 31104000000L;
 
+  // Delete max 20k records per job run
+  private static final int MAX_DELETE_PER_JOB_RUN = 20000;
+  // Delete any YamlChangeSets older than 30 days
+  private static final int RETENTION_PERIOD_IN_DAYS = 30;
+  // Delete 2k record in a batch
+  private static final String BATCH_SIZE = "500";
```
Comment (meenakshi-harness) [inline 400-rest/src/main/java/software/wings/yaml/gitSync/GitSyncEntitiesExpiryHandler.java:73]: this is 2k or 500?

**Review by boopesh007 — APPROVED** [review]
> Reviewed 1 of 2 files at r1, 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38738)* status: Some of the checks are too old.

**Review by abhinav-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38738)* status: 1 file(s) to review. 2. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`360-cg-manager/src/main/java/software/wings/app/WingsApplication.java` line 1487 at r2](https://reviewable.io/reviews/harness/harness-core/38738#-NED0FGOcxZJoLcbqh78-r2-1487:-NEbxRsNBvOCm6MtNZZg:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/76b1747b79614c8e5f2fd98c3fa37e0c69ea2aa0/360-cg-manager/src/main/java/software/wings/app/WingsApplication.java#L1487)):*
<details><summary><i>Previously, meenakshi-harness (Meenakshi Raikwar) wrote…</i></summary><blockquote>

Remove
</blockquote></details>

Done.
___
*[`400-rest/src/main/java/software/wings/yaml/gitSync/GitSyncEntitiesExpiryHandler.java` line 73 at r2](https://reviewable.io/reviews/harness/harness-core/38738#-NECtySjCZvJKjSU_a0S-r2-73:-NEbxSmO01cqBjO_1efX:b125p8o) ([raw file](https://github.com/harness/harness-core/blob/76b1747b79614c8e5f2fd98c3fa37e0c69ea2aa0/400-rest/src/main/java/software/wings/yaml/gitSync/GitSyncEntitiesExpiryHandler.java#L73)):*
<details><summary><i>Previously, meenakshi-harness (Meenakshi Raikwar) wrote…</i></summary><blockquote>

this is 2k or 500?
</blockquote></details>

blindly copied the logic from quartz. 
but it means to delete 20000 in one run for an account and delete 500 at once while deleting a batch of 20000

**Review by meenakshi-harness — APPROVED** [review]
> Reviewed 1 of 2 files at r1, 1 of 2 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38738)* status: :shipit: complete! ✓

___
*[`400-rest/src/main/java/software/wings/yaml/gitSync/GitSyncEntitiesExpiryHandler.java` line 73 at r2](https://reviewable.io/reviews/harness/harness-core/38738#-NECtySjCZvJKjSU_a0S-r2-73:-NEdEm-n0h4YIuNBj_Ew:bl3tti1) ([raw file](https://github.com/harness/harness-core/blob/76b1747b79614c8e5f2fd98c3fa37e0c69ea2aa0/400-rest/src/main/java/software/wings/yaml/gitSync/GitSyncEntitiesExpiryHandler.java#L73)):*
<details><summary><i>Previously, abhinav-harness (Abhinav Singh) wrote…</i></summary><blockquote>

blindly copied the logic from quartz. 
but it means to delete 20000 in one run for an account and delete 500 at once while deleting a batch of 20000
</blockquote></details>

Yes comment and value were not matching. We can update the comment later. Approving.

**Pull request merged**